### PR TITLE
Add awesome-promises

### PR DESCRIPTION
**Pull request by wbinnssmith**
Add an awesome curation of JavaScript Promise resources.

Not sure where this best belongs (maybe Misc. instead). Definitely a JavaScript resource, but belongs to both frontend and backend categories.

I'm open to feedback — let me know what you think of the list!

### CODE REVIEW

**Comment by sindresorhus** [comment]
Looks good.

-

Please read the [instructions](https://github.com/sindresorhus/awesome/blob/master/create-list.md) for creating a new list, [contributing guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [awesome manifest](https://github.com/sindresorhus/awesome/blob/master/awesome.md).

Especially this part:

> Wait at least a couple of weeks after creating a list before submitting to give it a chance to mature.

Ping me here in a few weeks and I'll reopen ;)

File: readme.md [hunk 1]
```diff
@@ -124,6 +124,7 @@ If you want to create your own list, please read the [create list instructions](
 - [CSS](https://github.com/sota0805/awesome-css)
 - [D3](https://github.com/wbkd/awesome-d3)
 - [Emails](https://github.com/jonathandion/awesome-emails)
+- [Promises](https://github.com/wbinnssmith/awesome-promises)
```
Comment (sindresorhus) [inline readme.md]: `Promises` => `JavaScript Promises`

Promises is too ambiguous as many languages supports them.

Comment (wbinnssmith) [inline readme.md]: agree :+1:

Comment (appleboy) [inline readme.md]: Agree :+1:

**Comment by wbinnssmith** [comment]
Hey @sindresorhus, it's been a couple of weeks and the list has evolved into a few more sections and has had some great contributions. Would you consider reopening this?

**Comment by sindresorhus** [comment]
Added ;)

**Pull request closed without merging**